src/twisted/persisted/dirdbm.py

```diff
@@ -60,120 +65,134 @@ def __init__(self, name):
             # but before renaming the replacement entry.
             #
             # NOTE: '.' is NOT in the base64 alphabet!
-            for f in glob.glob(os.path.join(self.dname, "*.new")):
+            for f in glob.glob(self._dnamePath.child("*.new").path):
                 os.remove(f)
-            replacements = glob.glob(os.path.join(self.dname, "*.rpl"))
+            replacements = glob.glob(self._dnamePath.child("*.rpl").path)
             for f in replacements:
                 old = f[:-4]
                 if os.path.exists(old):
                     os.remove(f)
                 else:
                     os.rename(f, old)
-    
+
+
     def _encode(self, k):
-        """Encode a key so it can be used as a filename.
+        """
+        Encode a key so it can be used as a filename.
         """
         # NOTE: '_' is NOT in the base64 alphabet!
-        return base64.encodestring(k).replace('\n', '_').replace("/", "-")
-    
+        return base64.encodestring(k).replace(b'\n', b'_').replace(b"/", b"-")
+
+
     def _decode(self, k):
-        """Decode a filename to get the key.
         """
-        return base64.decodestring(k.replace('_', '\n').replace("-", "/"))
-    
+        Decode a filename to get the key.
+        """
+        return base64.decodestring(k.replace(b'_', b'\n').replace(b"-", b"/"))
+
+
     def _readFile(self, path):
-        """Read in the contents of a file.
-        
+        """
```

Comment:
These are redundant with getContent/setContent. Since they are internal, they can just be killed, right?